### DISCUSSION TITLE
How to Set NLS_LANG  while connectting Oracle 11g?

### DISCUSSION THREAD

**Discussion by downmoon** [dbeaver/dbeaver#9075] [Q&A] [answered]
OS: windows 10 64bit
When I connect Oracle 11g,I get a SQL result. But data column‘s values are unintelligible code.
If I do this using  PL/sql developer,that result is OK. 

I use this prompt:
`set NLS_LANG=American_America.WE8ISO8859P1
start `plsqldev.exe``

Now,How can I do in DBeaver?

**Comment by rmrfchik** [comment]
Connection properties -> Oracle properties -> Session

**Comment by uslss** [comment]
Hi @downmoon ,
probably you could use bootstrap queries in connection properties
![image](https://user-images.githubusercontent.com/31996417/85410480-d4898100-b56f-11ea-9046-654f7218c153.png)

**Comment by downmoon** [comment]
> 
> 
> Connection properties -> Oracle properties -> Session

Hi,
thanks for your reply.   I select the same option according to the PL/sql develper,but it seems not properly.

`select userenv('language') from dual;`
`AMERICAN_AMERICA.WE8ISO8859P1`

![2020-06-24_10-46-50](https://user-images.githubusercontent.com/58096799/85492642-149c4280-b608-11ea-86ca-94911505bd01.png)

![2020-06-24_10-30-36](https://user-images.githubusercontent.com/58096799/85492078-2b8e6500-b607-11ea-9a14-73648c799d74.png)
![2020-06-24_10-33-23](https://user-images.githubusercontent.com/58096799/85492095-35b06380-b607-11ea-80f1-157a0a7aab40.png)

**Comment by downmoon** [comment]
> 
> 
> Hi @downmoon ,
> probably you could use bootstrap queries in connection properties


thanks.
 --Connection properties -> shell commands -> Before cnnection -> BAT: set NLS_LANG=AMERICAN_AMERICA.WE8ISO8859P1

It seems not work.

If I set this cmd  and  use Oracle's own sqlplus ,then result of query is OK.

**Comment by uslss** [comment]
@downmoon , and if you'll use Connection properties ->Initialization -> Bootstrap queries, does it solve the problem?
![image](https://user-images.githubusercontent.com/31996417/86589769-f125b000-bf96-11ea-87de-9e5f5117d40f.png)

**Comment by downmoon** [comment]
> 
> 
> @downmoon , and if you'll use Connection properties ->Initialization -> Bootstrap queries, does it solve the problem?
It's not going to work.
maybe,"set NLS_LANG=AMERICAN_AMERICA.WE8ISO8859P1" is a windows command,not a valid SQL command.

**Comment by downmoon** [comment, marked as answer]
I have resolved the problem with another tools--druid.

![dbeaver_004](https://user-images.githubusercontent.com/58096799/87753845-1c7d8a00-c836-11ea-9dbb-9b53efb9249a.png)

download：https://repo1.maven.org/maven2/com/alibaba/druid/1.1.23/

Tips：https://www.cnblogs.com/downmoon/p/13329324.html

**Reply by Teiemo** [reply]
I have a problem for Korean word broken, so I find way for few days..

I think this solution just for chinese right?
